x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/BasicDistributedJobsIT.java

```diff
@@ -302,7 +303,7 @@ public void testMaxConcurrentJobAllocations() throws Exception {
             Runnable r = () -> {
                 try {
                     internalCluster()
-                            .stopRandomNode(settings -> settings.getAsBoolean(MachineLearning.ML_ENABLED.getKey(), false));
+                            .stopRandomNode(settings -> DiscoveryNode.hasRole(settings, MachineLearning.ML_ROLE) == false);
```

Comment:
I think this line is supposed to match nodes where ML *is* enabled, no? The preceding comment says "stopping ml nodes".